lib/game.rb

```diff
@@ -110,27 +129,48 @@ def score
 		end
 	end
 
+	def win(player)
+		puts "#{player.name} won"
+		player.score+=1
+	end
+
+	def move(player)
+		symbol = player.symbol
+		puts "#{player.name} move (print a number from 0 to #{board.dim-1} and a letter from A to #{ALPHABET[board.dim-1]}) or 'Q' to quit\n"
+		player_input = gets.gsub(/\s+/, "").split('')
+		if board.valid_input?(player_input)
+			move_index = board.dim*(player_input[0].to_i)+ALPHABET.index(player_input[1].upcase)
+			if board.empty_space?(move_index)
+				board.change_matrix(player.symbol, move_index)
+			else
+				puts 'Invalid move'
+				move(player)
+			end
+		elsif player_input.length == 1 && player_input[0].upcase == 'Q'
+			@wanna_quit = true
+		else
+			puts 'Please make a valid move'
+			move(player)
+		end
+	end	
+
 	def play
-		until player_won? || board_full?
+		current_player = @player1
+		until @board.player_won?(@player1.symbol) || @board.player_won?(@player2.symbol) || @board.full? || @wanna_quit
```

Comment:
Do we need to check against a specific player? For the game to end we just need to check if there is a sequence of symbols in a row, a column or a diagonal. Once we know there is a sequence we can determine the winner based on the symbol or the current turn. 

Checking for a specific player symbol twice is not optimal